src/tensor/diag.jl

```diff
@@ -31,13 +31,13 @@ Base.eltype(::Type{<:Diag{T}}) where {T} = eltype(T)
 # Deal with uniform Diag conversion
 Base.convert(::Type{<:Diag{T}},D::Diag) where {T} = Diag{T}(data(D))
 
-Base.similar(D::Diag{T}) where {T} = Diag{T}(similar(data(D)))
```

Comment:
Don't we still want this version?

Reply:
Do we? The tests passed without it...

Reply:
Hmm interesting. I think we still want it anyway though. The goal is for the subtypes of `TensorStorage` to act like `AbstractVector` (I am thinking of defining `TensorStorage` that way). My guess is that this version failed for `delta` since `data(D)` is just a number, we can have a branch here for `NonuniformDiag` and `UniformDiag`.

Reply:
OK, I'll add it back :)